core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -11491,4 +11463,196 @@ public Sql schema(CalciteAssert.SchemaSpec schemaSpec) {
           relFn, transforms);
     }
   }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-7279">[CALCITE-7279]
+   * ClickHouse dialect should wrap nested JOINs</a>.
+   *
+   * <p>ClickHouse does not support nested JOIN syntax directly.
+   * When a JOIN's right side is another JOIN, it must be wrapped
+   * in a SELECT * FROM (...) subquery. */
+  @Test void testClickHouseNestedJoin() {
+    final String query = "SELECT e.empno, j.dname, j.loc\n"
+        + "FROM emp e\n"
+        + "LEFT JOIN (\n"
+        + "  SELECT d1.deptno, d1.dname, d2.loc\n"
+        + "  FROM dept d1\n"
+        + "  INNER JOIN dept d2 ON d1.deptno = d2.deptno\n"
+        + ") AS j ON e.deptno = j.deptno";
+
+    final String sql = sql(query)
+        .schema(CalciteAssert.SchemaSpec.JDBC_SCOTT)
+        .withClickHouse()
+        .exec();
+
+    assertThat(sql, containsString("LEFT JOIN (SELECT *"));
```

Comment:
I think use following is fine. Just using a expected string.
```
    sql(query)
        .schema(CalciteAssert.SchemaSpec.JDBC_SCOTT)
        .withClickHouse()
        .ok(expected);
```

Reply:
+1

Reply:
+1